lib/k8s-e2e-tests/src/lib.rs

```diff
@@ -75,28 +113,66 @@ pub fn make_test_pod_with_containers<'a>(
         spec: Some(PodSpec {
             containers,
             restart_policy: Some("Never".to_owned()),
+            affinity,
             ..PodSpec::default()
         }),
         ..Pod::default()
     }
 }
 
-pub fn make_test_pod<'a>(
+pub fn make_test_pod_with_affinity<'a>(
```

Comment:
I assume the node affinity requirement is because we use host volume? Probably good to add that as a comment to this function for future readers.

Reply:
Since the tests only scan the logs from an agent on a single node, we want to make sure that all the test pods are on the same node so the agent picks them all. I'll add a comment.